candy-machine/program/src/processor/mint.rs

```diff
@@ -0,0 +1,762 @@
+                //msg!("Moving by {}", eight_remainder);
+                if pos {
+                    index_to_use += eight_remainder;
+                } else {
+                    if index_to_use < 8 {
+                        break;
+                    }
+                    index_to_use -= reversed;
+                }
+            } else {
+                //msg!("Moving by 8");
+                if pos {
+                    index_to_use += 8;
+                } else {
+                    index_to_use -= 8;
+                }
+            }
+        } else {
+            let position_from_right = 7 - index_to_use
+                .checked_rem(8)
+                .ok_or(CandyError::NumericalOverflowError)?;
+            let mask = u8::pow(2, position_from_right as u32);
+
+            taken = mask & arr[my_position_in_vec];
+
+            match taken {
+                x if x > 0 => {
+                    //msg!("Index to use {} is taken", index_to_use);
+                    if pos {
+                        index_to_use += 1;
+                    } else {
+                        if index_to_use == 0 {
+                            break;
+                        }
+                        index_to_use -= 1;
+                    }
+                }
+                0 => {
+                    //msg!("Index to use {} is not taken, exiting", index_to_use);
+                    found = true;
+                    arr[my_position_in_vec] |= mask;
+                }
+                _ => (),
+            }
+        }
+    }
+    Ok((index_to_use, found))
+}
+
+pub fn get_config_line(
+    a: &Account<'_, CandyMachine>,
+    index: usize,
+    mint_number: u64,
+) -> Result<ConfigLine> {
+    if let Some(hs) = &a.data.hidden_settings {
+        return Ok(ConfigLine {
+            name: hs.name.clone() + "#" + &(mint_number + 1).to_string(),
+            uri: hs.uri.clone(),
+        });
+    }
+    msg!("Index is set to {:?}", index);
+    let a_info = a.to_account_info();
+
+    let mut arr = a_info.data.borrow_mut();
+
+    let (mut index_to_use, good) =
+        get_good_index(&mut arr, a.data.items_available as usize, index, true)?;
+    if !good {
+        let (index_to_use_new, good_new) =
+            get_good_index(&mut arr, a.data.items_available as usize, index, false)?;
+        index_to_use = index_to_use_new;
+        if !good_new {
+            return err!(CandyError::CannotFindUsableConfigLine);
+        }
+    }
+
+    msg!(
+        "Index actually ends up due to used bools {:?}",
+        index_to_use
+    );
+    if arr[CONFIG_ARRAY_START + 4 + index_to_use * (CONFIG_LINE_SIZE)] == 1 {
+        return err!(CandyError::CannotFindUsableConfigLine);
+    }
+
+    let data_array = &mut arr[CONFIG_ARRAY_START + 4 + index_to_use * (CONFIG_LINE_SIZE)
+        ..CONFIG_ARRAY_START + 4 + (index_to_use + 1) * (CONFIG_LINE_SIZE)];
+
+    let mut name_vec = vec![];
+    let mut uri_vec = vec![];
```

Comment:
Not sure how much it matters here but might be better to have one allocation and just do:
```
let mut name_vec = Vec::with_capacity(MAX_NAME_LENGTH);
let mut uri_vec = Vec::with_capacity(MAX_URI_LENGTH);
```

Reply:
I don't know about on the blockchain, but on my Macbook an empty Vec starts out with capacity of 8, and then reallocates to 16, 32, etc. as more items pushed, basically a standard dynamic array.